best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,285 @@
+
+If the EST Server returns a HTTP 200 response the certificate request was successful and the EST Client should use the returned TLS Certificate and its chain of trust for all further requests to its NMOS APIs. The EST Client MUST remove any previously issued TLS Certificates and key pairs.
+
+If the EST Server returns any other HTTP response, the request has been unsuccessful. The EST Client SHOULD re-submit the request to an alternative EST Server if present. Otherwise the EST Client SHOULD carry on using the existing TLS Certificate if still valid and re-attempt renewal at an appropriate point.
+
+### Expired Manufacturer Issued TLS Client Certificate
+
+If the Manufacturer issued TLS Client Certificate has expired or has been revoked, it MUST NOT be used by the EST Client for authentication. An EST Client MAY attempt to request a TLS Certificate following [Initial Certificate Provisioning](#initial-certificate-provisioning), without providing a TLS Client Certificate during the TLS handshake. If the EST Server supports manual authentication the request will be processed.
+
+If the EST Server fails to process the request the following actions MAY be taken:
+1. The EST Client MAY have a TLS Certificate for the target network manually installed on the device. This certificate MUST then be used for future [Certificate Renewals](#certificate-renewal)
+2. The Manufacturer issued TLS Client Certificate MAY be renewed by a software/firmware update, this update MUST contain a unique TLS certificate per device. The valid TLS Client Certificate MAY then be used during [Initial Certificate Provisioning](#initial-certificate-provisioning)
+
+### Connection to new network
+
+It MAY be desirable that when a EST Client is connected to a different network that it automatically requests a TLS Certificate for the new network. It MUST be possible to disable this functionality on the EST Client and provide a manual method for an authorised operator to initiate the [Initial Certificate Provisioning](#initial-certificate-provisioning) workflow.
+
+On start up or on change of network state the EST Client MUST attempt to discover the EST Server using [DNS-SD](#dns-sd-advertisement). The EST Client SHOULD make a request to the `/cacerts` endpoint, if the request is successful the EST Client should compare the returned Certificate to the currently install Root CA, if the Certificate is for a different Domain the EST Client MUST follow [Initial Certificate Provisioning](#initial-certificate-provisioning) workflow.
+
+## TODO:
+* Specification of returned TLS certificate format (eg, .p7, .pem)?
+* Specification of supported Cipher Suites and minimum key lengths?
+* Generate a new Key Pair for each TLS Certificate renewal?
+* Specification of Root CA rollover period and procedure?
+* Support for server side generation of keys?
+
```

Comment:
I'd suggest a TODO to think through whether we can allow use of the client certs in some comms with NMOS APIs as well as EST Servers. It'd be worth thinking this through fully in the context of BCP-003-02 now to avoid issues later.